shotover-proxy/src/tls.rs

```diff
@@ -80,6 +81,15 @@ impl TlsConnector {
 
         Ok(ssl_stream)
     }
+
+    pub async fn connect(&self, tcp_stream: TcpStream) -> Result<SslStream<TcpStream>> {
+        let ssl = self.connector.configure()?.into_ssl("localhost")?;
```

Comment:
this into_ssl("localhost") looks suspicious.
https://docs.rs/openssl/latest/openssl/ssl/struct.ConnectConfiguration.html#method.into_ssl
From reading the docs, it seems this might break in a real non localhost scenario?

It wouldnt have been a problem on redis because hostname verification is disabled there.

If we find an issue and it needs a complex solution/investigation I would be happy to land what we have so far as it at least works in a subset of scenarios and get non-localhost connections working in a follow up PR.

Reply:
![image](https://user-images.githubusercontent.com/39339036/154405768-8fd1c2dd-9771-46bc-beb3-42f57f005447.png)

I'm not sure if that is the right path. Maybe we need to providing SSL connections with a hostname to connect to instead of an address.

Reply:
were you able to reproduce a problem by connecting to an instance that isnt on local host?

Reply:
Yes, trying to connect to with hostname "notlocalhost" creates a problem.

Reply:
I'll open an issue regarding TLS configuration. I think we will want an option for validating the hostname as well.